src/statsUtils.js

```diff
@@ -0,0 +1,84 @@
+const {createWriteStream} = require('fs');
+const {Readable} = require('stream');
+
+const _ = require('lodash');
+
+class StatsSerializeStream extends Readable {
+  constructor(stats) {
+    super();
+    this._indentLevel = 0;
+    this._stringifier = this._stringify(stats);
+  }
+
+  get _indent() {
+    return '  '.repeat(this._indentLevel);
+  }
+
+  _read() {
+    let readMore = true;
+
+    while (readMore) {
+      const {value, done} = this._stringifier.next();
+
+      if (done) {
+        this.push(null);
+        readMore = false;
+      } else {
+        readMore = this.push(value);
+      }
+    }
+  }
+
+  * _stringify(obj) {
+    if (typeof obj === 'string' || typeof obj === 'number' || typeof obj === 'boolean' || obj === null) {
+      yield JSON.stringify(obj);
+    } else if (_.isArray(obj)) {
```

Comment:
Do we need to use lodash here or would built-in `Array.isArray` work, too?

```suggestion
    } else if (Array.isArray(obj)) {
```

Reply:
Fixed